jib-core/src/main/java/com/google/cloud/tools/jib/cache/Cache.java

```diff
@@ -55,6 +58,22 @@ private Cache(CacheStorage cacheStorage) {
     this.cacheStorage = cacheStorage;
   }
 
+  /**
+   * Saves an image manifest and container configuration.
+   *
+   * @param imageReference the image reference the save the manifest and container configuration for
+   * @param manifestTemplate the manifest
+   * @param containerConfigurationTemplate the container configuration
+   * @throws IOException if an I/O exception occurs
+   */
+  public void writeMetadata(
+      ImageReference imageReference,
+      ManifestTemplate manifestTemplate,
+      ContainerConfigurationTemplate containerConfigurationTemplate)
```

Comment:
Have you thought about the Docker manifest V2.1 case? There's no separate container config file, and the config is embedded in the manifest. Maybe it's better to have different method signatures for V2.1 and V2.2 (+ OCI)? (I haven't looked into OCI manifest format, but I guess that's very similar to Docker V2.2 and doesn't need special handling?)

Reply:
See https://github.com/GoogleContainerTools/jib/issues/1641#issuecomment-484187937 as an example of V2.1.

Reply:
Good point, I think it makes sense to have 2 signatures (`(ImageReference, BuildableManifestTemplate, ContainerConfigurationTemplate)` and `(ImageReference, V21ManifestTemplate)`?)

Reply:
Actually, maybe it'd make sense to just make `containerConfigurationTemplate` `@Nullable`, and if it's null it fills in for the default os/arch and skips writing the configuration file?

Reply:
Nevermind, it's redundant to write out the config if it exists in the V2.1 manifest. Just going to go with 2 signatures.

Reply:
I think `@Nullable` can work. It would be good to think about the corresponding signature for reading the files.

Reply:
Ah, didn't see your edit before I put the last comment.